operator/pkg/controllers/master/kubeconfigs.go

```diff
@@ -0,0 +1,153 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package master
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+
+	"github.com/awslabs/kit/operator/pkg/apis/infrastructure/v1alpha1"
+	"github.com/awslabs/kit/operator/pkg/errors"
+	"github.com/awslabs/kit/operator/pkg/utils/object"
+	"github.com/awslabs/kit/operator/pkg/utils/secrets"
+	"go.uber.org/zap"
+
+	"gopkg.in/yaml.v2"
+	v1 "k8s.io/api/core/v1"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	certutil "k8s.io/client-go/util/cert"
+)
+
+// reconcileKubeConfigs creates required kube configs and stores them as secret in api server
+func (c *Controller) reconcileKubeConfigs(ctx context.Context, controlPlane *v1alpha1.ControlPlane) error {
+	for _, request := range []*secrets.Request{
+		kubeAdminCertConfig(controlPlane.ClusterName()),
+		kubeSchedulerCertConfig(controlPlane.ClusterName()),
+		kubeControllerManagerCertConfig(controlPlane.ClusterName()),
+	} {
+		if err := c.reconcileConfigFor(ctx, request, controlPlane); err != nil {
+			return err
+		}
+	}
+	zap.S().Debugf("[%v] Kube configs reconciled", controlPlane.ClusterName())
+	return nil
+}
+
+func (c *Controller) reconcileConfigFor(ctx context.Context, request *secrets.Request, controlPlane *v1alpha1.ControlPlane) error {
```

Comment:
Consider inverting controlplane and the secret request. My vague sense of why is that the control plane is the higher order object/concept, so it should come first.